### ISSUE TITLE
Heap memory increased in network thread

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#6595]
### Describe the bug

Memory kept increasing and then got stable. So it's similar to a leak but it did not keep increasing forever

<img width="1324" alt="Screenshot 2024-03-26 at 09 08 30" src="https://github.com/ChainSafe/lodestar/assets/10568965/e1c9dc28-f71a-48a1-9018-6dbb7871bbcb">

This does not happen on any holesky nodes, and does not happen on test mainnet node.

### Expected behavior

Heap memory should be stable

### Operating system

Linux

### Lodestar version or commit hash

v1.17.0

**Comment by twoeths** [comment]
network thread `gc` time keeps increasing along with heap memory

<img width="847" alt="Screenshot 2024-03-26 at 09 16 22" src="https://github.com/ChainSafe/lodestar/assets/10568965/9cbd7f41-d13a-44fd-90b6-1cb8ed9433c9">

`gc` happens randomly and if it runs right at block proposal time (when network thread need to work hard to serialize big block message) then it may cause a delay in #6596

**Comment by twoeths** [comment]
it looks more like a memory leak with 100 peers in @nflaig node https://github.com/ChainSafe/lodestar/pull/6556#issuecomment-2017692440

**Comment by nflaig** [comment]
> it looks more like a memory leak with 100 peers in @nflaig node [#6556 (comment)](https://github.com/ChainSafe/lodestar/pull/6556#issuecomment-2017692440)

yeah, it didn't look like it was stopping to increase on my node, still collecting data with 50 peers but will switch to 100 peers again and take some heap snapshots

**Comment by twoeths** [comment]
link to the heap snapshot https://github.com/ChainSafe/lodestar/pull/6556#issuecomment-2043250725

**Comment by twoeths** [comment]
tracked in this [libp2p issue](https://github.com/libp2p/js-libp2p/issues/2477)

**Comment by twoeths** [comment]
need to try `@libp2p/tcp` v9.0.22 which is a candidate for this issue, see https://github.com/libp2p/js-libp2p/pull/2487

right now I get this error updating it

```
@lodestar/beacon-node: src/network/libp2p/index.ts:76:7 - error TS2322: Type '(components: TCPComponents) => Transport' is not assignable to type '(components: Components) => Transport'.
@lodestar/beacon-node:   Property '[transportSymbol]' is missing in type 'import("/Users/tuyennguyen/Projects/workshop/lodestar/packages/beacon-node/node_modules/@libp2p/interface/dist/src/transport/index").Transport' but required in type 'import("/Users/tuyennguyen/Projects/workshop/lodestar/node_modules/@libp2p/interface/dist/src/transport/index").Transport'.
@lodestar/beacon-node:  76       tcp({
@lodestar/beacon-node:           ~~~~~
@lodestar/beacon-node:  77         maxConnections: networkOpts.maxPeers,
@lodestar/beacon-node:     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@lodestar/beacon-node: ... 
@lodestar/beacon-node:  85         },
@lodestar/beacon-node:     ~~~~~~~~~~
@lodestar/beacon-node:  86       }),
@lodestar/beacon-node:     ~~~~~~~~
@lodestar/beacon-node:   ../../node_modules/@libp2p/interface/dist/src/transport/index.d.ts:53:5
@lodestar/beacon-node:     53     [transportSymbol]: true;
@lodestar/beacon-node:            ~~~~~~~~~~~~~~~~~
@lodestar/beacon-node:     '[transportSymbol]' is declared here.
@lodestar/beacon-node: Found 1 error in src/network/libp2p/index.ts:76
```

@wemeetagain could you help with this? 🙏

**Comment by wemeetagain** [comment]
still happening

![Screenshot from 2024-10-22 16-33-21](https://github.com/user-attachments/assets/cd66752f-9a60-463e-b87d-bcc11a1e64d0)

**Comment by philknows** [comment]
Could also be worth investigating alongside #7186 and #7188
